jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1158,13 +1197,20 @@ public Method getDeclaredMethod(String name, Class<?>... parameterTypes) throws
 /**
  * Answers the class which declared the class represented
  * by the receiver. This will return null if the receiver
- * is a member of another class.
+ * is not a member of another class.
  *
  * @return		the declaring class of the receiver.
  */
 @CallerSensitive
 public Class<?> getDeclaringClass() {
-	Class<?> declaringClass = getDeclaringClassImpl();
+	if (cachedDeclaringClassOffset == -1){
+		cachedDeclaringClassOffset = getFieldOffset("cachedDeclaringClass");
+	}
+	if (cachedDeclaringClass == Dummy.class) {
+		writeFieldValue(cachedDeclaringClassOffset, getDeclaringClassImpl());//$NON-NLS-1$
```

Comment:
`//$NON-NLS-1$` should be with `cachedDeclaringClass`, pls add a space before it.